### PR TITLE
Feature/update download copy and design enhancements

### PR DESCRIPTION
**Pull request by mr-nick17**
### What

- Updated download copy (copied from [ticket](https://trello.com/c/4uiSQBWr/5644-download-component-content-changes))
```
- XLSX (0.0 KB)
Dataset and supporting information
- CSV (0.0 KB)
Machine readable dataset
- CSVW (0.0 KB)
Machine readable supporting information
- JSON (0.0 KB)
- TXT (0.0 KB)
Supporting information
```
- Converting to renderer table of contents UI component
- Other minor design enhancements

### How to review

- Sense check
- Tests pass
- Look at image ⬇️ 
- Alternatively, you can pull the branch and run locally by following the [cantabular import](https://github.com/ONSdigital/dp-compose/tree/main/cantabular-import#readme) instructions or call me 🤙 and I can show you locally. 
get data and table of contents component

### Who can review

Frontend dev

### Dependencies

- [x]  ~Waiting for [PR](https://github.com/ONSdigital/dp-renderer/pull/80) to be approved first~

### CODE REVIEW

File: assets/templates/partials/census/get-data.tmpl [hunk 1]
```diff
@@ -20,29 +20,33 @@
...
-                                    </span>
-                                </label>
-                            </span>
+                            <label class="ons-radio__label{{ if or (eq .Extension "txt") (eq .Extension "csvw") (eq .Extension "xls") }} ons-label--with-description{{ end }}" for="{{ .Extension }}" id="{{ .Extension }}-label"><span class="ons-u-tt-u">{{ .Extension }}</span> format ({{ humanSize .Size }})
+                                <span id="{{ .Extension }}-label-description-hint" class="ons-label__description ons-radio__label--with-description">
+                                    {{ if or (eq .Extension "xls") (eq .Extension "xlsx") }}
+                                        {{- localise "IncludesSupportingInfo" $.Language 1 -}}
+                                    {{ else if eq .Extension "csv" }}
+                                        {{- localise "MachineReadable" $.Language 1 -}}&nbsp;<span class="ons-u-tt-l">{{- localise "Dataset" $.Language 1 -}}</span>
+                                    {{ else if eq .Extension "csvw" }}
```
Comment (NoelleDL) [inline assets/templates/partials/census/get-data.tmpl:29]: Is the &nbsp intentional? If so, wouldn't padding/margin be better here?

File: assets/templates/partials/census/get-data.tmpl [hunk 2]
```diff
@@ -20,29 +20,33 @@
...
+                                        {{- localise "MachineReadable" $.Language 1 -}}&nbsp;<span class="ons-u-tt-l">{{- localise "SupportingInfo" $.Language 1 -}}</span>
+                                    {{ else if eq .Extension "txt" }}
```
Comment (NoelleDL) [inline assets/templates/partials/census/get-data.tmpl:31]: Same here.

File: assets/templates/partials/census/get-data.tmpl (hunk 1 above)
Comment (mr-nick17) [reply assets/templates/partials/census/get-data.tmpl:29]: I put it in to make the code a bit more readable and (hopefully) prevent the space from being removed accidentally. Happy to replace it with normal whitespace if you'd prefer

Comment (NoelleDL) [reply assets/templates/partials/census/get-data.tmpl:29]: I think normal whitespace would be better!

File: mapper/mapper_test.go [hunk 3]
```diff
@@ -794,6 +794,18 @@ func TestCreateCensusDatasetLandingPage(t *testing.T) {
 		So(page.DatasetLandingPage.IsFlexible, ShouldBeTrue)
 		So(page.DatasetLandingPage.FormAction, ShouldEqual, fmt.Sprintf("/datasets/%s/editions/%s/versions/%s/filter-flex", flexDm.ID, versionOneDetails.Edition, strconv.Itoa(versionOneDetails.Version)))
 	})
+
+	Convey("Downloads on version, HasDownloads bool is set correctly", t, func() {
+		page := CreateCensusDatasetLandingPage(context.Background(), req, pageModel, oneContactDetailDM, dataset.Version{Downloads: nil}, datasetOptions, dataset.VersionDimensions{}, "", false, []dataset.Version{}, 1, "", "", 50, false)
```
[the remark this answers is not in the document]
Comment (mr-nick17) [reply mapper/mapper_test.go:799]: I could nest the convey statements. Something like:
```go
Convey("Test HasDownloads"... {
  Convey("HasDownloads set to true when downloads greater than zero" ...)
  Convey("HasDownloads set to false when downloads count zero" ...)
})
```

[the remark this answers is not in the document]
Comment (NoelleDL) [reply mapper/mapper_test.go:799]: I think this is better.

**Review by NoelleDL — APPROVED** [review]

**Pull request merged**